lib/Adafruit_MPR121/Adafruit_MPR121.cpp

```diff
@@ -56,8 +57,16 @@ boolean Adafruit_MPR121::begin(uint8_t i2caddr) {
   writeRegister(MPR121_FDLT, 0x00);
 
   writeRegister(MPR121_DEBOUNCE, 0);
-  writeRegister(MPR121_CONFIG1, 0x10); // default, 16uA charge current
-  writeRegister(MPR121_CONFIG2, 0x20); // 0.5uS encoding, 1ms period
+  switch(_sensitivity) {
+    case MPR212_EXTRA_SENSITIVITY:
+      writeRegister(MPR121_CONFIG1, 0x20); // increased to 32uA charge current
+      writeRegister(MPR121_CONFIG2, 0x3A); // 0.5uS encoding, 18 samples, 4ms period
+      break;
+    default: // MPR212_NORMAL_SENSITIVITY
+      writeRegister(MPR121_CONFIG1, 0x10); // default, 16uA charge current
+      writeRegister(MPR121_CONFIG2, 0x20); // 0.5uS encoding, 1ms period
+      break;
```

Comment:
Just writing down my notes as I already had them to verify the comments.

![image](https://user-images.githubusercontent.com/3751318/147699948-6ca7ad4b-b14d-4034-8b40-517dbdc8a002.png)

Extra sensitivity:
conf1: 0x20 = 00 100000
- FFI = 00 = 6 samples  (default)
- CDC = 100000 = 32 = 32 uA
conf2: 0x3A = 001 11 010
- CDT = 001 = 0.5 usec (default)
- SFI = 11 = 18 samples
- ESI = 010 = 2 = 4 msec

Normal sensitivity:
conf1: 0x10 = 00 010000
- FFI = 00 = 6 samples  (default)
- CDC = 010000 = 16 = 16 uA  (default)
conf2: 0x20 = 001 00 000
- CDT = 001 = 0.5 usec (default)
- SFI = 00 = 4 samples (default)
- ESI = 000 = 0 = 1 msec

Used acronyms:
- CDC: global electrode charge/discharge current
- CDT: global charge/discharge time
- FFI: First Filter Iterations - The first filter iterations field selects the number of samples taken as input to the first level of filtering. 
- SFI: Second Filter Iterations - Selects the number of samples taken for the second level filter
- ESI: Electrode Sample Interval - Selects the period between samples used for the second level of filtering

To be honest, the nr of samples and cycle time seem quite short for something that is operated by a human.
So I wonder if you may also need to adapt the filtering for 50Hz and 60Hz regions.
Can't find it in the datasheet.
I can imagine you may also need to adapt the thresholds for some specific enclosures, as glass does interact different compared to ABS or other plastics.

Maybe there is something about it online giving some numbers from real world use cases?

Reply:
> Maybe there is something about it online giving some numbers from real world use cases?

All I have is from the linked issue:
> After making a change, the sensor works through 6mm glass without false alarms
> Information taken from here https://forums.adafruit.com/viewtopic.php?f=19&t=72025&start=15